.gitignore

```diff
@@ -1,9 +1,18 @@
+.idea
+.bloop
+.bsp
+.metals
+.scala
+.vscode
+
+target
+out
+metals.sbt
+*.iml
+
 *.class
 *.log
-target/
+.eclipse
+
 *~
-.idea
-out
 .DS_Store
-.bsp
-project
```

Comment:
Need to keep `project`, as IntelliJ will occasionally out-generate one.

Reply:
`sbt` has a `project` folder. So this won't be an issue because we are not using `sbt`. 

Also, I am not sure if you are referring to `project` or `.project` folder. Where does IJ create the `project` folder?

Reply:
InteliJ creates a `scala-commons/project` directory, which we don't want to commit because we're not using it.